ibis/sql/tests/test_compiler.py

```diff
@@ -2022,3 +2021,95 @@ def test_not_exists_anti_join_case(self):
   WHERE t0.key1 = t1.key1
 )"""
         assert result == expected
+
+
+class TestUDFStatements(unittest.TestCase):
+
+    def setUp(self):
+        self.con = MockConnection()
+        self.name = 'test_name'
+        self.inputs = ['string', 'string']
+        self.output = 'int64'
+
+    def test_create_udf(self):
+        stmt = ddl.CreateFunction('/foo/bar.so', 'testFunc', self.inputs,
+                                  self.output, self.name)
+        result = stmt.compile()
+        expected = ("CREATE FUNCTION test_name(string, string) returns bigint "
+                    "location '/foo/bar.so' symbol='testFunc'")
+        assert result == expected
+    
+    def test_create_udf_type_conversions(self):
+        stmt = ddl.CreateFunction('/foo/bar.so', 'testFunc',
+                                  ['string', 'int8', 'int16', 'int32'],
+                                  self.output, self.name)
+        result = stmt.compile()
+        expected = ("CREATE FUNCTION test_name(string, tinyint, smallint, int) "
+                    "returns bigint "
+                    "location '/foo/bar.so' symbol='testFunc'")
+        assert result == expected
+        
+        
+    def test_delete_udf_simple(self):
+        stmt = ddl.DropFunction(self.name, self.inputs, must_exist=True)
+        result = stmt.compile()
+        expected = "DROP FUNCTION IF EXISTS test_name(string, string)"
```

Comment:
the logic on `must_exist` is flipped here, `IF EXISTS` should only be there if `must_exist=False`. Can you add a test case that triggers this?